tests/installation/yast2_firstboot.pm

```diff
@@ -46,45 +47,50 @@ sub license {
     send_key $cmd{next};
 }
 
-sub welcome {
+sub firstboot_welcome {
     assert_screen 'welcome';
     wait_screen_change(sub { send_key $cmd{next}; }, 7);
 }
 
-sub clock_and_timezone {
+sub firstboot_timezone {
     assert_screen 'inst-timezone';
     wait_screen_change(sub { send_key $cmd{next}; }, 7);
 }
 
-sub user_setup {
-    my $is_not_shared_passwd = shift;
+sub firstboot_user {
     assert_screen 'local_user';
     enter_userinfo(username => get_var('YAST2_FIRSTBOOT_USERNAME'));
-    if (defined($is_not_shared_passwd)) {
-        send_key 'alt-t' if (is_opensuse);
-    }
-    else {
-        send_key 'alt-t' if (is_sle('>=12-sp4'));
-    }
     wait_screen_change(sub { send_key $cmd{next}; }, 7);
     await_password_check;
-    wait_screen_change(sub { send_key $cmd{next}; }, 7) unless defined($is_not_shared_passwd);
 }
 
-sub root_setup {
+sub firstboot_root {
     assert_screen 'root_user', 60;
     enter_rootinfo;
     wait_screen_change(sub { send_key $cmd{next}; }, 7);
 }
 
+sub firstboot_hostname {
+    if (check_screen 'bsc1173298') {
+        record_soft_failure "bsc#1173298";
+        send_key $cmd{ok};
+        wait_screen_change(sub { send_key $cmd{next}; }, 7);
+        return;
+    }
+    assert_screen 'hostname';
+    wait_screen_change(sub { send_key $cmd{next}; }, 7);
+}
+
 sub run {
-    my $self = shift;
-    language_and_keyboard;
-    welcome;
-    $self->license;
-    clock_and_timezone;
-    user_setup(1);
-    root_setup;
+    my $self      = shift;
+    my $test_data = get_test_suite_data();
+    my %clients;
+    foreach my $client (@{$test_data->{clients}}) {
+        # Make sure the subroutine called from test data exists
+        die "Client '$client' is not defined in the module, please check test_data" unless defined(&{"$client"});
+        my $client_method = \&{"$client"};
+        $client_method->($self);
```

Comment:
I think `$self` is passed always by default, but I might be wrong, does it work with `$client_method->();` ?